src/onnx/onnx.cpp

```diff
@@ -2112,21 +2116,25 @@ template <class... Ts>
 program parse_onnx_from(onnx_options options, Ts&&... xs)
 {
     onnx_parser parser;
-    parser.batch_size = options.batch_size;
-#ifndef NDEBUG
-    // Log the program when it can't be parsed
-    try
+    parser.batch_size             = options.batch_size;
+    parser.skip_unknown_operators = options.skip_unknown_operators;
+    if(options.print_program_on_error)
     {
-        parser.parse_from(std::forward<Ts>(xs)...);
+        // Log the program when it can't be parsed
+        try
```

Comment:
One way to get better coverage could be to call parse_onnx() in onnx_test.cpp with print_program_on_error=true.